### PR TITLE
Answer Issue #71: Auto sort list

### PR DESCRIPTION
**Pull request by mattb555**
Refactored the "Sort list" element to be more like the "Change view" element above it, added third sorting option (custom sort), Alphabetical Sort and Rank now disable click and drag and auto sort games in after each new insertion.

### CODE REVIEW

File: src/store/mutations.js [hunk 1]
```diff
@@ -145,6 +145,10 @@ export default {
         state.gameLists[state.platform.code][listIndex].view = view;
     },
 
+    UPDATE_LIST_SORT(state, {listIndex, sortOrder }) {
```
Comment (romancm) [inline src/store/mutations.js]:
```suggestion
    UPDATE_LIST_SORT(state, { listIndex, sortOrder }) {
```

File: src/components/GameBoard/ListSettings.vue [hunk 2]
```diff
@@ -45,23 +45,14 @@
...
+                v-for="(icon, sortOrder) in sortOrders"
+                :key="sortOrder"
+                class="small primary"
+                :class="{ hollow: activeList.sortOrder !== sortOrder }"
+                @click="setListSort(sortOrder)"
             >
-                <i class="fas fa-sort-numeric-up" />
-                {{ $t('list.sortByRating') }}
+                <i :class="icon" />
+                {{ $t('list.' + sortOrder) }}
```
Comment (romancm) [inline src/components/GameBoard/ListSettings.vue]:
```suggestion
                {{ $t(`list.${sortOrder}`) }}
```

File: src/components/GameBoard/ListSettings.vue [hunk 3]
```diff
@@ -182,6 +181,25 @@ export default {
...
+            this.$store.commit('UPDATE_LIST_SORT', {
+                listIndex: this.activeListIndex,
+                sortOrder,
+            });
+            this.sort(sortOrder);
+            this.$emit('update');
+        },
+
+        sort(sortOrder) {
+            if (sortOrder == 'sortByName') {
```
Comment (romancm) [inline src/components/GameBoard/ListSettings.vue]:
```suggestion
            if (sortOrder === 'sortByName') {
```

File: src/components/GameBoard/ListSettings.vue [hunk 4]
```diff
@@ -182,6 +181,25 @@ export default {
...
+                this.$store.commit('SORT_LIST_ALPHABETICALLY', this.activeListIndex);
+                this.$emit('update', 'List sorted alphabetically');
+            } else if (sortOrder == 'sortByRating') {
```
Comment (romancm) [inline src/components/GameBoard/ListSettings.vue]:
```suggestion
            } else if (sortOrder === 'sortByRating') {
```

File: src/components/GameCards/GameCardCover.vue [hunk 5]
```diff
@@ -6,7 +6,7 @@
             @click="openDetails"
         >
 
-        <i class="fas fa-grip-vertical game-drag-handle" />
+        <i v-bind:class="{ 'fas fa-grip-vertical game-drag-handle' : list.sortOrder == 'sortByCustom' || list.sortOrder == null }" />
```
Comment (romancm) [inline src/components/GameCards/GameCardCover.vue]: I think you should use `v-if` here so the item doesn't render at all if it's not manually sortable
```suggestion
<i
    v-if="list.sortOrder == 'sortByCustom' || list.sortOrder == null"
    class="fas fa-grip-vertical game-drag-handle"
/>
```

also if template logic gets too long consider moving it to a computed prop like `isDraggable` or similar

```suggestion
<i
    v-if="isDraggable"
    class="fas fa-grip-vertical game-drag-handle"
/>
```

File: src/components/GameCards/GameCardDefault.vue [hunk 6]
```diff
@@ -5,7 +5,7 @@
         <div class="game-info">
             <a v-text="game.name" @click="openDetails" v-if="list.view !== 'covers'" />
 
-            <i class="fas fa-grip-vertical game-drag-handle" />
+            <i v-bind:class="{ 'fas fa-grip-vertical game-drag-handle' : list.sortOrder == 'sortByCustom' || list.sortOrder == null }" />
```
Comment (romancm) [inline src/components/GameCards/GameCardDefault.vue]: same here, you can put the computed prop in `GameCard.js` so it's shared across all gamecard components, hope that makes sense

File: src/components/GameCards/GameCardWide.vue [hunk 7]
```diff
@@ -9,7 +9,8 @@
                 @click="openDetails"
             />
 
-            <i class="fas fa-grip-vertical game-drag-handle" />
+            <i v-bind:class="{ 'fas fa-grip-vertical game-drag-handle' : list.sortOrder == 'sortByCustom' || list.sortOrder == null }" />
+
```
Comment (romancm) [inline src/components/GameCards/GameCardWide.vue]: same here, also looks like you have an extra line here

File: src/components/GameBoard/ListSettings.vue [hunk 8]
```diff
@@ -157,6 +153,9 @@ export default {
 
     mounted() {
         this.localList = JSON.parse(JSON.stringify(this.activeList));
+        this.$bus.$on('GAMES_ADDED', () => {
+            this.sort(this.activeList.sortOrder);
+        });
```
Comment (romancm) [inline src/components/GameBoard/ListSettings.vue]: make sure you turn shut down the listener on `beforeDestroy`
```
    beforeDestroy() {
        this.$bus.$off('GAMES_ADDED');
    },

**Comment by romancm** [comment]
@all-contributors please add @mattb555 for code

**Comment by mattb555** [comment]
I have made all requested changes

**Pull request merged**